packages/caravan-fees/src/rbf.ts

```diff
@@ -0,0 +1,303 @@
+import { PsbtV2 } from "@caravan/psbt";
+import BigNumber from "bignumber.js";
+import { DEFAULT_DUST_THRESHOLD, RBF_SEQUENCE } from "./constants";
+import { RbfOptions, FeeRate } from "./types";
+
+/**
+ * RbfTransaction class for handling Replace-By-Fee transactions
+ */
+class RbfTransaction {
+  private psbt: PsbtV2;
+  private newFeeRate: FeeRate;
+  private dustThreshold: string;
+
+  /**
+   * Create a new RbfTransaction
+   * @param psbt - The original transaction as a PsbtV2
+   * @param newFeeRate - The new fee rate for the replacement transaction
+   * @param options - Additional options for RBF
+   *   - dustThreshold: The minimum amount (in satoshis) for an output to not be considered dust.
+   *If not provided, a default value is used.
+   */
+  constructor(
+    psbt: PsbtV2,
+    newFeeRate: FeeRate,
+    options?: { dustThreshold?: string }
+  ) {
+    this.psbt = new PsbtV2(psbt.serialize("base64"));
+    this.newFeeRate = newFeeRate;
+    this.dustThreshold =
+      options?.dustThreshold || DEFAULT_DUST_THRESHOLD.toString();
```

Comment:
Just make DEFAULT_DUST_THRESHOLD a string to start with. Convert to a big number on class initialization. Inside the class you're probably going to prefer to keep things in their more "complex" states (e.g. using the PsbtV2 instance, BigNumber operations, etc.). Whenever the class takes args or exports though it should be operating in "universal" primitives (strings, numbers, etc.) which decouples the package from requiring other dependencies.

Reply:
Great ... understood 👍